runtime/src/main/java/org/corfudb/runtime/view/AbstractView.java

```diff
@@ -118,6 +118,8 @@ public Layout getCurrentLayout() {
                         Thread.sleep(runtime.getParameters().getConnectionRetryRate().toMillis());
                     } catch (InterruptedException e) {
                         log.warn("Interrupted Exception in layout helper.", e);
+                        Thread.currentThread().interrupt();
+                        throw new RuntimeException(e);
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Define and throw a dedicated exception instead of using a generic one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00112)

Reply:
done.

Reply:
Done.